### ISSUE TITLE
[AI] Infra: GPU Hang / Reset (1 job)

### ISSUE THREAD

**Issue by dhonnappa-amd** [dhonnappa-amd/build-dashboards-testing#4028]
## AI Analysis — Infrastructure Issue: GPU Hang / Reset

**1 affected job**

| Pipeline/Build | Queue | Job Link |
|---------------|-------|----------|
| amd-ci/7148 | `amd_mi355_1` | [View job](https://buildkite.com/vllm/amd-ci/builds/7148#019d4508-2b52-4fff-b526-f87994917dba) |

---

### Summary

This job was running vLLM attention backend correctness tests on AMD ROCm GPUs, specifically testing the flex_attention backend with sliding window encoder configurations. The test successfully loaded models, initialized the GPU environment, and began executing test_sliding_window_encoder_backend_correctness. During test execution, PyTorch's torch.compile attempted to compile the attention forward pass using the Inductor backend, which in turn invoked the Triton compiler for AMD GPUs. The Triton AMD backend crashed with a segmentation fault at line 336 in make_llir (LLIR generation phase) while compiling kernels for the flex_attention block mask building operation. This is a GPU compiler crash, not a test logic failure. The segfault occurred deep in the Triton->LLVM compilation stack for ROCm, indicating either a compiler bug, incompatible GPU architecture assumptions, or memory corruption during kernel compilation. This appears to be a real regression or incompatibility issue rather than infrastructure flakiness, as the crash is deterministic and occurs at a specific compilation phase.

---

### Root Cause Analysis

Walking through the log chronologically: The job successfully completed all setup phases (git checkout, Docker image pull, GPU reset, environment configuration). ROCm system checks passed showing clean GPU state with no existing processes. The test suite began execution and successfully loaded the Gemma3TextModel for attention testing. The first unrecoverable error occurred at 21:05:29 when Python encountered a segmentation fault during Triton AMD backend compilation. The stack trace shows the crash happened in triton/backends/amd/compiler.py:336 in make_llir, which is the LLIR (LLVM IR) generation phase for AMD GPUs. This was triggered by torch.compile attempting to optimize the flex_attention forward pass for a sliding window encoder test. The segfault is the root cause - there were no prior errors, no retries, and the process was immediately killed with signal 11 (SIGSEGV). This is a GPU compiler crash specific to the Triton AMD backend when compiling attention kernels.

---

### Detected Events (all affected jobs)

**amd-ci/7148** (queue: `amd_mi355_1`):
- `gpu_hang`: Fatal Python error: Segmentation fault

Thread 0x00007b58c0625000 (most recent call first):
  File "/usr/local/lib/python3.12/dist-packages/triton/backends/amd/compiler.py", line 336 in make_llir
  - _Segmentation fault occurred during Triton AMD backend compilation (make_llir), indicating a GPU compiler crash or memory access violation during kernel compilation for ROCm_
- `gpu_hang`:   File "/vllm-workspace/tests/v1/attention/test_attention_backends.py", line 732 in test_sliding_window_encoder_backend_correctness
  File "/usr/local/lib/python3.12/dist-packages/vllm/v1/attention/backends/flex_attention.py", line 680 in build_block_mask
  File "/usr/local/lib/python3.12/dist-packages/vllm/v1/attention/backends/flex_attention.py", line 971 in forward
  - _The crash occurred during attention backend testing with flex_attention, specifically when building block masks for sliding window attention, triggering torch.compile -> Triton compilation_
- `gpu_hang`:   File "/usr/local/lib/python3.12/dist-packages/torch/_inductor/compile_fx.py", line 2486 in compile_fx
  File "/usr/local/lib/python3.12/dist-packages/triton/compiler/compiler.py", line 324 in compile
  File "/usr/local/lib/python3.12/dist-packages/triton/backends/amd/compiler.py", line 487 in <lambda>
  - _PyTorch Inductor was compiling a graph using Triton for AMD GPUs when the segfault occurred in the LLIR generation phase of the Triton AMD compiler_
- `other:exit status 139`: 🚨 Error: The command exited with status 139
user command error: exit status 139
  - _Exit code 139 = 128 + 11 (SIGSEGV), confirming the process was killed by a segmentation fault signal_

---

#### Immediate Fix

1. Disable torch.compile for this specific test by adding @torch.compiler.disable or setting PYTORCH_DISABLE_INDUCTOR=1. 2. Skip the flex_attention backend tests on ROCm until the Triton compiler issue is resolved. 3. Add pytest.mark.skip decorator to test_sliding_window_encoder_backend_correctness with reason='Triton AMD compiler segfault in make_llir'.

#### Suggested Code Change

```
In tests/v1/attention/test_attention_backends.py:

@pytest.mark.skipif(
    platform.system() == 'Linux' and 'rocm' in torch.version.hip,
    reason='Triton AMD backend segfaults in make_llir during flex_attention compilation'
)
def test_sliding_window_encoder_backend_correctness(...):
    ...

OR add to the test file:

import torch._dynamo
torch._dynamo.config.suppress_errors = True

OR in the specific test:

with torch.compiler.disable():
    run_attention_backend(...)
```

#### Prevention

1. Add CI job that tests Triton compilation separately before running full test suite to catch compiler crashes early. 2. Pin Triton version and add compatibility matrix for ROCm versions in requirements. 3. Add timeout and retry logic specifically for torch.compile operations with fallback to eager mode. 4. Report this segfault to Triton AMD backend maintainers with full stack trace and ROCm version (6.16.6). 5. Add telemetry to track which Triton kernels fail compilation on different ROCm versions. 6. Consider adding TRITON_CACHE_DIR to persistent storage to avoid recompilation and reduce crash frequency.

#### Workaround

Set environment variable PYTORCH_DISABLE_INDUCTOR=1 before running tests to bypass torch.compile and use eager mode execution. This will make tests slower but avoid the Triton compiler crash. Alternatively, use VLLM_ATTENTION_BACKEND=XFORMERS or VLLM_ATTENTION_BACKEND=FLASH_ATTN to bypass flex_attention entirely for this test run.

---

_This issue was automatically created by the AI log analysis pipeline._
